### PR TITLE
Robot Socket implementation

### LINKED ISSUE #45: RobotSocket implementation
[`RobotSocket`](https://github.com/Comet-Robotics/chessBot/blob/main/src/server/robot/robotsocket.ts) is an object of the Robot that abstracts robot calls into TCP messages that interface with the physical robot. To accomplish this, you will need to port code from the old implementation of the chess bot server, from which the connection manager can be found in [BotServer.js @ commit `263e34`](https://github.com/Comet-Robotics/chessBot/blob/263e34affd7d03fb54ba061efd2c0ad32cc3bb32/webServer/include/BotServer.js). The `drive` and `turn` methods should be implemented, and any other pertinent messages to or from the robot should be exposed as public functions as well. The RobotSocket should also handle reconnecting to a robot if connection is lost.

### PR DESCRIPTION
**Pull request by democat3457**
Fixes #45 

Separates WebSocket behavior into a new class, BotTunnel, which represents a socket connection. Based on the old BotServer implementation.

Additionally, adds a react router and a debug page for sending basic commands to the robot, and upgrades vite to fix vulnerabilities.

### CODE REVIEW

**Comment by xXDMOGXx** [comment]
Still needs to be properly tested

File: src/server/api/api.ts [hunk 1]
```diff
@@ -67,3 +128,24 @@ function makeMove(move: { from: string; to: string }) {
 function processMove(from: Square, to: Square): Command {
   throw new Error("Function not implemented");
 }
+
+function manualMove(robotId: number, leftPower: number, rightPower: number): boolean {
+  if (!(tcpServer.getConnectedIDs().includes(robotId.toString()))) {
```
Comment (AlexKempen) [inline src/server/api/api.ts:133]: This function uses both `ID` and `Id`. I personally vote in favor of standardizing around `Id` - my understanding is that's what google recommends as well.

**Review by AlexKempen — CHANGES_REQUESTED** [review]
Some minor things to tweak.

File: src/client/app.tsx [hunk 2]
```diff
@@ -1,11 +1,19 @@
 import { FocusStyleManager } from "@blueprintjs/core";
+import { BrowserRouter as Router, Route, Routes } from "react-router-dom";
```
[the remark this answers is not in the document]
Comment (democat3457) [reply src/client/app.tsx:2]: https://stackoverflow.com/a/74091353/8846453
seems to be pretty standard

[the remark this answers is not in the document]
Comment (AlexKempen) [reply src/client/app.tsx:2]: Sure, fine. It'll be moot anyways if we switch to v6 syntax

**Review by AlexKempen — APPROVED** [review]

**Pull request merged**